api/models/FixedDateRange.ts

```diff
@@ -15,4 +15,34 @@ export const FixedDateRangeSchema = z.optionalParseable(
     
     return new FixedDateRange(parsedArg.startDateTime, parsedArg.endDateTime)
   }
-)
\ No newline at end of file
+)
+
+export const MIN_EVENT_DURATION = 60
+
+/**
+ * A Zod schema for creating a FixedDateRange with additional constraints for new Events:
+ * - Duration must be at least MIN_EVENT_DURATION seconds.
+ * - The end date cannot be in the past.
+ */
+export const CreateFixedDateRangeSchema = FixedDateRangeSchema.superRefine((dateRange, ctx) => {
+  const { startDateTime, endDateTime } = dateRange;
+
+  const secondDiff = (endDateTime.getTime() - startDateTime.getTime()) / 1000;
+  const isEndDatePast = endDateTime < new Date();
```

Comment:
We may want to revisit this check in the future to ensure that the right timezones are being compared between client and server.

Reply:
I'll add a comment